test/test_reader.py

```diff
@@ -67,3 +68,17 @@ def test_loads_all_coordinates(self):
         npt.assert_equal(data.coordinates.x, self.expected_xs)
         npt.assert_equal(data.coordinates.y, self.expected_ys)
         npt.assert_equal(data.coordinates.z, self.expected_zs)
+
+class TestLoadImzML(unittest.TestCase):
```

Comment:
I'd like to see any mocked out imzML reading pipeline tested. You can go for some `patch`ing and mock out the true reader. This way you will be able to test just your pipeline, without the underlying data (only on the data you provide through `patch`).

Reply:
I need to do some research on that.

Reply:
What you can do is e.g.: `with patch.object(pyimzml.PyImzMLParser, '__enter__') as mocked_parser:` before the call to `load_imzml`. This way, in the `load_imzml` variable `input_handle` becomes a mock and you can substitute different things through it into the code (i.e. coordinates, mzs and spectra).

The only problematic thing may be the way you import `pymzml.PyImzMLParser` into `reader.py` - in current manner it is *copied* into `reader` module (construct `from ... import ...`). If you switch to *linking* to the definition (construct `import X as Y` and then calling `Y.PyImzMLParser`), it will still point to the module it is defined in.